nuxeo-compound-documents/src/main/java/org/nuxeo/compound/documents/CompoundDocumentServiceImpl.java

```diff
@@ -72,25 +102,29 @@ protected boolean isAllowedEntry(ZipEntry entry) {
         return !ignoredFiles.ignore(path);
     }
 
-    protected void createEntry(DocumentModel compoundDoc, String archiveName, ZipFile zip, ZipEntry entry) {
+    protected void createEntry(DocumentModel compoundDoc, ZipFile zip, ZipEntry entry, String compoundFolderType) {
         Path entryPath = new Path(entry.getName());
         Path entryParentPath = entryPath.removeLastSegments(1);
         String parentDocPath = compoundDoc.getPath().append(entryParentPath).toString();
         String entryDocName = entryPath.lastSegment();
         var session = compoundDoc.getCoreSession();
         if (entry.isDirectory()) {
-            var doc = session.createDocumentModel(parentDocPath, entryDocName, COMPOUND_FOLDER_DOCTYPE);
+            var doc = session.createDocumentModel(parentDocPath, entryDocName, compoundFolderType);
             session.createDocument(doc);
         } else {
             var blob = new ZipEntryBlob(zip, entry);
             blob.setFilename(entryDocName);
             FileImporterContext ctx = FileImporterContext.builder(session, blob, parentDocPath).build();
             try {
                 Framework.getService(FileManager.class).createOrUpdateDocument(ctx);
-            } catch (IOException e) {
+            } catch (NuxeoException | IOException e) {
                 String message = String.format(
-                        "Failed to create CompoundDocument: %s. something went wrong with archive: %s at entry: %s.",
-                        compoundDoc.getName(), archiveName, entry.getName());
+                        "Failed to create CompoundDocument: %s. Something went wrong with archive: %s at entry: %s.",
+                        compoundDoc.getName(), zip.getName(), entry.getName());
+                if (e instanceof NuxeoException) {
+                    var statusCode = ((NuxeoException)e).getStatusCode();
+                    throw new NuxeoException(message, e, statusCode);
```

Comment:
Use `NuxeoException#addInfo` here:
```suggestion
                    ((NuxeoException) e).addInfo(message);
                    throw e;
```

Reply:
I cast the thrown e to not add a `throws Exception` : 
```
                if (e instanceof NuxeoException) {
                    ((NuxeoException) e).addInfo(message);
                    throw (NuxeoException) e;
                }
```
Or maybe a local variable ```NuxeException ne = (NuxeoException) e;``` would be better ?

Reply:
Different catch clauses would be better, just need to factorize the message :)
if you have any idea to do it @kevinleturc 😬

Reply:
Not really.

Reply:
So let store the cast in a var instead of casting it 2 times.